audio-worklet/src/bitcrusher-demo.js

```diff
@@ -124,40 +129,26 @@ class BitcrusherDemo {
           name: 'Volume'
         });
 
-    // GUI controls allow switching between audio worklet and script processor.
-    // The script processor node is used by default, and switching is not
-    // enabled if the browser does not support audio worklet.
-    if (this.useWorkletAndScriptProcessor_) {
-      this.useScriptProcessorCheckBox_ = document.createElement('input');
-      this.useScriptProcessorCheckBox_.type = 'checkbox';
-      this.useScriptProcessorCheckBox_.checked = true;
-      const audioworkletLabel = document.createElement('label');
-      audioworkletLabel.textContent =
-          'Use ScriptProcessor (if unchecked, use AudioWorklet Bitcrusher).';
-
-      this.useScriptProcessorCheckBox_.addEventListener(
-          'change', this.switchBitcrusherNode.bind(this));
-
-      let container = document.getElementById(containerId);
-      container.appendChild(this.useScriptProcessorCheckBox_);
-      container.appendChild(audioworkletLabel);
-    }
+    let workletButton = document.getElementById(workletButtonId);
+    if (this.useWorklet_)
+      workletButton.addEventListener('click', this.workletSelected.bind(this));
+    else workletButton.disabled = true;
+    
+    document.getElementById(scriptProcessorButtonId)
+        .addEventListener('click', this.scriptProcessorSelected.bind(this));
   }
 
-  /**
-   * Switch between whether the audio worklet node or script processor node
-   * connects to the master gain node.
-   */
-  switchBitcrusherNode() {
-    // The function is called after the checkbox switches state.
-    const changeToAudioWorklet = !this.useScriptProcessorCheckBox_.checked;
-    if (changeToAudioWorklet) {
-      this.bitcrusherAudioWorklet_.connect(this.masterGain_);
-      this.bitcrusherScriptProcessor_.output.disconnect();
-    } else {
-      this.bitcrusherScriptProcessor_.output.connect(this.masterGain_);
-      this.bitcrusherAudioWorklet_.disconnect();
-    }
+  workletSelected() {
+    // Switching occurs 10ms into the future for thread synchronization.
+    this.workletGain_.gain.setValueAtTime(1, this.context_.currentTime + 0.01);
+    this.scriptProcessorGain_.gain.setValueAtTime(
+        0, this.context_.currentTime + 0.01);
```

Comment:
Because `currentTime` can advance between the two calls to `setValueAtTime`, you probably want to do something like:
```
let t = context.currentTime + 0.01;
workletGain.gain.setValueAtTime(1, t);
scriptGain.gain.setValueAtTime(0,t);
```